### DISCUSSION TITLE
how to run process asynchronously with tm1py?

### DISCUSSION THREAD

**Discussion by fxmenard** [cubewise-code/tm1py#1059] [General]
**Describe what did you try to do with TM1py**
I have a big number of processes to run on a TM1 instance. I want to launch these processes asynchronously to execute them in parallel. 

**Describe what's not working the way you expect**
In TurboIntegrator I would use RunProcess rather than ExecuteProcess. In tm1py it seems it is the same method processes.execute_with_return, but passing parameter "return_async_id=True". Unfortunately, the processes keep running one after the other. Any idea??

**Version**
- TM1py [e.g. 1.11.3] 
- TM1 Server Version: [e.g. 11.4]

**Additional context**
Code chunk:

``` python
with open( 'myCsvFileWith40000Lines', newline='') as csvfile:
                paramReader = csv.reader(csvfile, delimiter=',', quotechar='"')
                for param in paramReader:
                    logging.info( f"running {param}" )
                    sTM1Server.processes.execute_with_return(
                        process_name=param[0],
                        return_async_id=True,
                        dim1=param[2],
                        dim2=param[4],
                        dim3=param[6],
                        dim4=param[8],
                        pValue=float( param[10] )
                    )
```

**Comment by MariusWirtz** [comment]
Have you considered using RushTI?
https://github.com/cubewise-code/rushti

**Reply by fxmenard** [reply]
Hello Marius,
Thanks for the quick reply. I will try to use RushTI, hadn't noticed its existence. It's a bit less convenient for me than using a library for this, but it should do the trick. looks like there are quite a few useful options such as the number of parallel threads allowed for a given execution.
